core/rmutex.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2016 Theobroma Systems Design & Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    core_sync Synchronization
```

Comment:
this needs to be "ingroup" instead of defgroup. The group is defined in mutex.h.